File: additionalProblem/week27/week27_oh/BestSet.cpp

#include <string>
#include <vector>

using namespace std;

vector<int> solution(int n, int s) {
    vector<int> ans;

    int t = s / n;
    int k = s % n;

    if (n > s) {
        ans.push_back(-1);
        return ans;
    }

    for (int i = 0; i < n; i++)
        ans.push_back(t);
    for (int i = n - 1; i > n - 1 - k; i--)
        ans[i]++;

    return ans;
}

File: additionalProblem/week27/week27_jeongHee/BestSet.java
package stack;

public class BestSet {
    //https://school.programmers.co.kr/learn/courses/30/lessons/12938
    //최고의 집합
    public int[] solution(int n, int s) {
        int[] ans = new int[n];
        if (s / n == 0) {
            return new int[]{-1};
        }
        for (int i = 0; i < n - s % n; i++) {
            ans[i] = (s / n);
        }
        for (int i = n - s % n; i < n; i++) {
            ans[i] = (s / n + 1);
        }
        return ans;
    }

}
